src/snapred/ui/widget/WorkflowNode.py

```diff
@@ -0,0 +1,36 @@
+from snapred.ui.model.WorkflowNodeModel import WorkflowNodeModel
+from snapred.ui.presenter.WorkflowNodePresenter import WorkflowPresenter
+from snapred.ui.view.WorkflowNodeView import WorkflowNodeView
+
+
+def startWorkflow(action, subview):
+    return WorkflowNodeModel(action=action, view=subview, nextModel=None)
+
+
+def continueWorkflow(workflow, action, subview):
+    currentWorkflow = workflow
+    while currentWorkflow.nextModel is not None:
+        currentWorkflow = workflow.nextModel
```

Comment:
This doesn't look like it does the correct thing.  This looks like it'd be an infinite loop, unless there's something else happening here

Reply:
Michael wrote this as part of the approval UI workflow, I'm not sure how works when executed.

Reply:
@walshmm

Reply:
@rboston628 `nextModel` is `None` by default, this is just iterating a linked list of workflow objects